spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/DefaultAccessTokenConverter.java

```diff
@@ -147,21 +183,25 @@ public OAuth2Authentication extractAuthentication(Map<String, ?> map) {
 		if (includeGrantType && map.containsKey(GRANT_TYPE)) {
 			parameters.put(GRANT_TYPE, (String) map.get(GRANT_TYPE));
 		}
-		Set<String> resourceIds = new LinkedHashSet<String>(map.containsKey(AUD) ? getAudience(map)
+		Set<String> resourceIds = new LinkedHashSet<String>(map.containsKey(AUD) ? extractAudience(map)
 				: Collections.<String>emptySet());
 		
-		Collection<? extends GrantedAuthority> authorities = null;
-		if (user==null && map.containsKey(AUTHORITIES)) {
-			@SuppressWarnings("unchecked")
-			String[] roles = ((Collection<String>)map.get(AUTHORITIES)).toArray(new String[0]);
-			authorities = AuthorityUtils.createAuthorityList(roles);
-		}
+		Collection<? extends GrantedAuthority> authorities = user == null ? extractRequestAuthorities(map) : null;
 		OAuth2Request request = new OAuth2Request(parameters, clientId, authorities, true, scope, resourceIds, null, null,
 				null);
 		return new OAuth2Authentication(request, user);
 	}
 
-	private Collection<String> getAudience(Map<String, ?> map) {
+	protected Collection<GrantedAuthority> extractRequestAuthorities(Map<String, ?> map) {
+		if (map.containsKey(AUTHORITIES)) {
+			@SuppressWarnings("unchecked")
+			String[] roles = ((Collection<String>) map.get(AUTHORITIES)).toArray(new String[0]);
+			return AuthorityUtils.createAuthorityList(roles);
+		}
+		return null;
+	}
+
+	protected Collection<String> extractAudience(Map<String, ?> map) {
```

Comment:
We cannot rename `getAudience` to `extractAudience` in a patch or minor release. This can only happen in a major release as we follow [Semantic Versioning](https://semver.org/spec/v2.0.0.html). Also, we do not plan on delivering a 3.0 release. I'm not sure you are aware but please see this [announcement](https://spring.io/blog/2018/01/30/next-generation-oauth-2-0-support-with-spring-security).